test/e2e/vmagent_test.go

```diff
@@ -71,9 +73,134 @@ func TestVMAgent(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Error when setting up VMAgent test: %v", err)
 	}
+	k8sUtils, err = NewKubernetesUtils(data)
+	failOnError(err, t)
 	defer teardownVMAgentTest(t, data, vmList)
 	t.Run("testExternalNode", func(t *testing.T) { testExternalNode(t, data, vmList) })
 	t.Run("testExternalNodeWithANP", func(t *testing.T) { testExternalNodeWithANP(t, data, vmList) })
+	t.Run("testExternalNodeSupportBundleCollection", func(t *testing.T) { testExternalNodeSupportBundleCollection(t, data, vmList) })
+}
+
+func (data *TestData) waitForDeploymentRealized(t *testing.T, namespace string, name string, timeout time.Duration) error {
+	t.Logf("Waiting for Deployment '%s/%s' to be realized", namespace, name)
+	if err := wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		dp, err := data.clientset.AppsV1().Deployments(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return dp.Status.ObservedGeneration == dp.Generation && dp.Status.ReadyReplicas == *dp.Spec.Replicas, nil
+	}); err != nil {
+		return fmt.Errorf("error when waiting for Deployment '%s/%s' to be realized: %v", namespace, name, err)
+	}
+	return nil
+}
+
+func (data *TestData) waitForSupportBundleCollectionRealized(t *testing.T, name string, timeout time.Duration) error {
+	t.Logf("Waiting for SupportBundleCollection '%s' to be realized", name)
+	var sbc *crdv1alpha1.SupportBundleCollection
+	if err := wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		sbc, conditionErr := data.crdClient.CrdV1alpha1().SupportBundleCollections().Get(context.TODO(), name, metav1.GetOptions{})
+		if conditionErr != nil {
+			return false, conditionErr
+		}
+		return sbc.Status.DesiredNodes == sbc.Status.CollectedNodes, nil
```

Comment:
This condition is not enough. When the support bundle is not processed at all, the number will also equal.

Reply:
updated